src/v/compat/cluster_compat.h

```diff
@@ -350,6 +350,12 @@ struct compat_check<cluster::topic_properties> {
         json_write(record_value_subject_name_strategy_compat);
         json_write(initial_retention_local_target_bytes);
         json_write(initial_retention_local_target_ms);
+        if (obj.mpx_virtual_cluster_id) {
+            json::write_member(
+              wr,
+              "mpx_virtual_cluster_id",
+              ssx::sformat("{}", obj.mpx_virtual_cluster_id.value()()));
+        }
```

Comment:
Suggestion:  You can add:
```c++
inline void rjson_serialize(
  json::Writer<json::StringBuffer>& w, const model::vcluster_id& v) {
    rjson_serialize(w, ssx::sformat("{}", v));
}
```

to `compat/json.h` and then you can maintain the same `json_write(mpx.virtual_cluster_id)` pattern